diff --git a/packages/protocol/contracts/L1/libs/LibProposing.sol b/packages/protocol/contracts/L1/libs/LibProposing.sol
index d5d9a2814..7963acb7d 100644
--- a/packages/protocol/contracts/L1/libs/LibProposing.sol
+++ b/packages/protocol/contracts/L1/libs/LibProposing.sol
@@ -136,7 +136,7 @@ library LibProposing {
             // 7645: Alias ORIGIN to SENDER
             if (
                 _config.checkEOAForCalldataDA
-                    && ECDSA.recover(meta_.blobHash, local.params.signature) != msg.sender
+                    && ECDSA.recover(meta_.blobHash, local.params.signature) != tx.origin
             ) {
                 revert L1_INVALID_SIG();
             }
diff --git a/packages/protocol/contracts/L1/tiers/TierProviderBase.sol b/packages/protocol/contracts/L1/tiers/TierProviderBase.sol
index ca45e5fe1..d14c2d714 100644
--- a/packages/protocol/contracts/L1/tiers/TierProviderBase.sol
+++ b/packages/protocol/contracts/L1/tiers/TierProviderBase.sol
@@ -20,8 +20,8 @@ abstract contract TierProviderBase is ITierProvider {
                 verifierName: "",
                 validityBond: 125 ether, // TKO
                 contestBond: 250 ether, // TKO
-                cooldownWindow: 1440, //24 hours
-                provingWindow: 15, // 15 minutes
+                cooldownWindow: 6, // for local test: 24hours --> 6 min
+                provingWindow: 1, // for local test: 15minutes --> 1 minutes
                 maxBlocksToVerifyPerProof: 0
             });
         }
diff --git a/packages/protocol/package.json b/packages/protocol/package.json
index 04325eaa5..e72f960ac 100644
--- a/packages/protocol/package.json
+++ b/packages/protocol/package.json
@@ -8,6 +8,7 @@
     "compile": "forge build --build-info --extra-output storage-layout",
     "layout": "./deployments/gen-layouts.sh",
     "test:deploy": "./script/download_solc.sh && ./script/test_deploy_on_l1.sh",
+    "test:local_deploy": "./script/download_solc.sh && ./script/deploy_on_l1.sh",
     "eslint": "pnpm exec eslint --fix --ignore-path .eslintignore --ext .js,.ts .",
     "fmt:sol": "forge fmt",
     "generate:genesis": "ts-node ./utils/generate_genesis/main.ts",
diff --git a/packages/relayer/docker-compose/docker-compose.yml b/packages/relayer/docker-compose/docker-compose.yml
index 66192fde6..c0b50f897 100644
--- a/packages/relayer/docker-compose/docker-compose.yml
+++ b/packages/relayer/docker-compose/docker-compose.yml
@@ -1,13 +1,13 @@
 version: "3.8"
 services:
-  db:
+  db1:
     image: mysql:8.0
     cap_add:
       - SYS_NICE
     restart: always
     environment:
       - MYSQL_DATABASE=relayer
-      - MYSQL_ROOT_PASSWORD=<password>
+      - MYSQL_ROOT_PASSWORD=12345678
     ports:
       - "3306:3306"
     volumes:
@@ -55,7 +55,7 @@ services:
       - prometheus
 
 volumes:
-  db:
+  db1:
     driver: local
   prometheus_data: {}
   grafana_data: {}
